### ISSUE TITLE
Routing via building entrances?

### ISSUE THREAD

**Issue by neiljp** [graphhopper/graphhopper#336]
From the performance of graphhopper on current OSM data (ie. the new functionality on osm.org), it seems that routing does not go from building entrances, ie. entrance=* or building=entrance, but rather simply from the centre of a building?

While for car routing this is not an issue (being very zoomed out?), for foot routing it is often useful to know entrances and routes.

I'm currently looking at a University campus, for example:
http://www.openstreetmap.org/directions?engine=graphhopper_foot&route=57.16568%2C-2.10589%3B57.16478%2C-2.10386#map=18/57.16520/-2.10471

**Comment by karussell** [comment]
But if you put the marker there it will route you there, so is this probably more a geocoding issue?

**Comment by ChaoticMind** [comment]
Nah, it doesn't take the entrance as an exit when you move the market inside the building. It only appears to do so when the closest road extremity to your marker becomes the footpath north of the building (in the example above).

This is done regardless of the presence of any entrances.

It just means distance(marker, footpath) < distance(marker, service_road_middle_of_building).

**Comment by karussell** [comment]
GraphHopper does not know anything about the building, to know which entrance is close GH would need the indoor map or at least the area which would increase hardware resources etc a lot. But maybe we find a cheaper solution.

**Comment by karussell** [comment]
Hmmh, but this service road in the middle of the building has no entries into the building? Are there different kinds of entries so that e.g. a main entry should be preferred?

**Comment by ChaoticMind** [comment]
@karussell you mean in general or in that example? In general, [yes](http://wiki.openstreetmap.org/wiki/Key:entrance), but in this example it's unspecified. But I think I didn't really understand the question.

**Comment by karussell** [comment]
This might be a duplicate of #371? (or as this is older the other way around but the new issue has pretty images ;))
